fix: avoid exception by possible undefined event

calhaudev opened the pull request:
When a user fills out a validated form and hits ENTER in the keyboard, the lib throws an exception by undefined event in onChange handler. This PR just verifies if the event exists.

![image](https://github.com/react-hook-form/react-hook-form/assets/17502027/68ebf952-8d4e-4c1f-a844-a75f8a516fa5)

calhaudev committed:
fix: avoid exception by possible undefined event

calhaudev committed:
fix: avoid exception by possible undefined event

calhaudev committed:
Merge branch 'master' into fix/form-control-onchange-handler

calhaudev committed:
Merge branch 'react-hook-form:master' into fix/form-control-onchange-handler

calhaudev committed:
Merge branch 'master' into fix/form-control-onchange-handler

rafaelcalhau commented:
hi @bluebill1049, my first time trying to contribute here. This PR is very simple, is there another requirement for this to be merged, please let me know.

bluebill1049 committed:
🐞 fix #11218 `useFieldArray` focus issue with rules prop (#11221)

* 🐞 fix #11218 useFieldArray focus issue with rules prop

* fix issue with not focus

bluebill1049 committed:
🐞 fix #11229 issue with values not re-render the root form (#11231)

ssi02014 committed:
🍢 improved `getDirtyFieldsFromDefaultValues` type (#11238)

thomasdbock committed:
📮 fix(#11239): Set fields disabled state based on form and field disabled (#11241)

* fix(#11239): Set fields disabled state based on form and field disabled prop

* fix(#11239): If-check and unit test

* Update createFormControl.ts

---------

Co-authored-by: Thomas De Bock <thomas.de.bock@persgroep.net>
Co-authored-by: Beier (Bill) <bluebill1049@hotmail.com>

mildfuzz committed:
⌨️ change type structure of `UseFormHandleSubmit` (#11245)

* Change signature of UseFormHandleSubmit

* add api-extractor report

Lennon57 committed:
🔧 refactor(util function):  Enhance Consistency in Utility Functions: Standardizing Export, Function Types, and Naming (#11268)

* refactor: Matching get and set util conventions

- obj => object
- set util change arrow function

* refactor: change sleep util

- change arrow function
- change export to export default
- The import of the file that uses the function has been modified.

* refactor: change deepMerge util

- change export to export default
- The import of the file that uses the function has been modified.

* refactor: api-extractor.md

- The changed get function argument is reflected in the report

* refactor: change arrow function

* Revert "refactor: api-extractor.md"

This reverts commit e5ff061b9bef52774d669aff4e58c75024af61f7.

* fix: api-extractor.md build

* Revert "fix: api-extractor.md build"

This reverts commit d64ec64015830d36db0d3820939d4bd7043cdfa2.

* fix: check api-extractor build error

bluebill1049 committed:
fix #11267 disabled prop not affect controller validation (#11273)

bluebill1049 committed:
7.49.0

bluebill1049 committed:
🥼 update changelog v7.49.0

bluebill1049 committed:
🐞 fix #11276 issue with disabled omit on Controller (#11277)

bluebill1049 committed:
🐞 fix #11287 disabled prop cause const dirtyFields update (#11291)

* 🐞 fix #11287 disabled prop cause const dirtyFields update

* fix issue with test

jantimon committed:
🩺 add test for disabled form field submit (#11288)

* add test for disabled form field submit

* fix lint

---------

Co-authored-by: Beier (Bill) <bluebill1049@hotmail.com>

jantimon committed:
🩺  add broken test for useController (#11290)

Co-authored-by: Beier (Bill) <bluebill1049@hotmail.com>

bluebill1049 committed:
7.49.1

bluebill1049 committed:
🐞 fix #11292 issue with controller validation prop (#11294)

* fix #11292 issue with controller valdiation prop

* remove log

jantimon committed:
🩺 add broken test for useController (#11293)

Co-authored-by: Beier (Bill) <bluebill1049@hotmail.com>

bluebill1049 committed:
7.49.2

geeee committed:
🐞 fix #11328 resetField with object as defaultValue (#11329)

RulerOfCakes committed:
🖨️ fix: Change `DeepPartial` type definition to be compatible with `unknown` (#11333)

* fix: DeepPartial type definition

* fix: api-extractor build

Moshyfawn committed:
🐛 fix(useController): check if disabled state was set by user (#11353)

* 🐛 fix(useController): check if disabled state was set by user

Closes: #11320

* ✅ test(useController): disabled state is not overriden if not set

Related: #11320

* 🎨 style(useController): format whitespaces

* 🐛 fix(useController): set disabled state if passed to controller

* ✅ test(useController): should respect disabled state on controller

* 🎨 style: run prettier

Moshyfawn committed:
🐛 fix(reset): remove global reset option flag (#11355)

Closes: #11308

Moshyfawn committed:
🐛 fix(formState, disabled): infer controlled disabled form state (#11361)

Closes: #11360

bluebill1049 committed:
7.49.3

jonahallibone committed:
🐞 fix #11374: Nested deep partial with unknown (#11373)

* fix: Nested deep partial with unknown

* reset editor files

* added api extractor

bluebill1049 committed:
🐞 fix #11368 keep form state mounted when keeping values (#11384)

* fix #11368 keep form state mounted when keeping values

* fix logic condition

* update unit tests

* remove only

* improve unit test

bluebill1049 committed:
🧑‍✈️ fix #11281 issue with upgrade node version (#11392)

eladattias1996 committed:
⌨️ fix #10734 watch and useWatch without parameters return type (#11359)

Co-authored-by: Beier (Bill) <bluebill1049@hotmail.com>

bluebill1049 committed:
💩 fix #11322 trigger extra re-render with useWatch plus shouldUnregister (#11394)

bluebill1049 committed:
🐞 close #11246 skip root validation on action when mode and revalidate mode is onSubmit (#11395)

* 🐞 close #11246 skip root validation on action when mode and revalidatemode is onSubmit

* fix lint error

Moshyfawn committed:
✨ feat(values): priorities values over defaultvalues (#11399)

Closes: #11398

RulerOfCakes committed:
⌨️ fix: change `DeepPartial` to account for object union types & `unknown` (#11385)

* feat: extract objects from DeepPartial check

* chore: api-extractor

* fix: api-extractor build

---------

Co-authored-by: Beier (Bill) <bluebill1049@hotmail.com>

anonkey committed:
💫 feat(useForm): leave formState flags in a consistent state when throwing in handleSubmit onValid (#11214)

Co-authored-by: Beier (Bill) <bluebill1049@hotmail.com>

r-kataoka11 committed:
🏷️ improve `CustomElement` type (#11309)

* 🏷️ improve CustomElement type

* 🏷️ Change HTMLElement to Partial

* 📝 fix api-extractor

---------

Co-authored-by: Beier (Bill) <bluebill1049@hotmail.com>

IanVS committed:
🐞 fixes #11386 get new dirtyFields when reset keeping defaults and dirty values (#11401)

* Add test

* Get newly-dirty fields when keeping defaults and dirty values

Moshyfawn committed:
👷 build(Engines): remove pnpm dependency (#11425)

Closes: #11424

Ar4ys committed:
🔧 fix data param in handleSubmit when TTransformedValues is a union type (#11420)

* Rewritten UseFormHandleSubmit to use default values in generics

* Updated api-extractor report

---------

Co-authored-by: Beier (Bill) <bluebill1049@hotmail.com>

bluebill1049 committed:
7.50.0

ssi02014 committed:
✅ chore: add noop Utils Function (#11455)

* feat: Add Noop Util Function

* test: Add Noop Test Code

* fix: Fixed TestCode

bluebill1049 committed:
Revert "⌨️ fix #10734 watch and useWatch without parameters return type (#11359)"

This reverts commit fdbadbb7a613d272223cce14c9610484103ce1f6.

bluebill1049 committed:
7.50.1

bluebill1049 committed:
🐞 close #11503 issue dirty field update with async validation (#11509)

deepgolani4 committed:
🧽 chore: removed unused utils and resp. test cases (#11476)

* Removed unused utils and resp. test cases

* remove unused functions

* added removed code

---------

Co-authored-by: deep.golani <deep.golani@bfhl.in>

bluebill1049 committed:
❤️ thank you principal.com for your contribution

bluebill1049 committed:
Revert "🔧 fix data param in handleSubmit when TTransformedValues is a union type (#11420)"

This reverts commit 6fee03caa768dc612f884f209cf4dea2ba7eb701.

bluebill1049 committed:
🤡 fix flush root render hack (#11521)

* 🤡 fix flush root render hack

* fix preitter issue

bluebill1049 committed:
Revert "✨ feat(values): priorities values over defaultvalues (#11399)" (#11545)

This reverts commit a0583ff60c777d9070c770c31af4cbed00176a80.

bluebill1049 committed:
😾 fix #11514 issue under strict mode reset invoke twice reset mounted field names with keepDirtyValues (#11546)

pbankonier committed:
💬 add `isValidating` and `validatingFields` to field state (#10657)

* Add isValidating to field state

* Fix type tests for isValidating field state

* Adjust bundlewatch maxSize

* Fix api-extractor.md merge fail

* Adjust formState validatingFields property

* Adjust bundlewatch maxSize

* Refactoring for isValidating fieldState

* Fix multi async validators behavior

* Remove unnecessary explicit type definition in isValidating test

* Improve _updateIsValidating method

* Add proxyFormState check to updateIsValidating method

---------

Co-authored-by: Beier (Bill) <bluebill1049@hotmail.com>

bluebill1049 committed:
💅 related #11541 improve dirty check when `undefined` provided with `reset` api

* related #11541 improve dirty check when undefined provided with reset api

* early exit when no values provided

bluebill1049 committed:
🧗‍♂️ chore: update lib dev deps (#11548)

deepgolani4 committed:
⌨️ chore: replace ts-ignore with ts-expect-error (#11573)

* chore: replace ts-ignore with ts-expect-error

* chore: replace ts-ignore with ts-expect-error

---------

Co-authored-by: deep.golani <deep.golani@bfhl.in>

bluebill1049 committed:
7.51.0

bluebill1049 committed:
🥼 update changelog v7.51.0

bluebill1049 committed:
🐞 fix #11590 issue with trigger on validatingFields (#11613)

* 🐞 fix #11590 issue with trigger on validatingFields

* fix unit tests and lint error

Moshyfawn committed:
💚 ci: cleanup actions versions (#11629)

* 💚 ci(Node): bump node action version

* 💚 ci(Issues): bump lock issue action version

Moshyfawn committed:
🐛 bug(validatingFields, trigger): handle all fields validation trigger (#11624)

* WIP 🐛 bug(validatingFields, trigger): handle all fields validation trigger

* 🔧 chore: update lock file

* 🐛 bug(validatingFields, trigger): handle all fields validation trigger

* 🔧 chore: remove todo comment

* fix #11621 issue with undefined name as field name

* fix lint

* revert change on trigger

---------

Co-authored-by: bill <bluebill1049@hotmail.com>

bluebill1049 committed:
7.51.1

bluebill1049 committed:
fix #11719 validation stuck on true state (#11723)

* fix #11719 validation stuck on true state

* fix tests

* fix test

bluebill1049 committed:
7.51.2

bluebill1049 committed:
🐞 fix #11773 regression on dirty check with reset (#11775)

bluebill1049 committed:
7.51.3

bluebill1049 committed:
🩻 update README.md (#11776)

bluebill1049 committed:
👹 close #11778 improve unregister omit key with getValues method (#11779)

bluebill1049 committed:
7.51.4

mjr2595 committed:
📖 fix broken link to examples in README.md (#11805)

Co-authored-by: Beier (Bill) <bluebill1049@hotmail.com>

cj-young committed:
⌨️ close: correct type of `error` field in `getFieldState` return object (#11831)

* fix: correct type of `error` field in `getFieldState` return object

* fix: update reports

calhaudev committed:
Merge branch 'master' into fix/form-control-onchange-handler
